### ISSUE TITLE
Mouse not functioning correctly with the latest update (3.14.29)

### ISSUE THREAD

**Issue by 4nexus5** [ValveSoftware/gamescope#1467]
I updated my gamescope to the latest version yesterday and after updating my mouse did everything except work correctly in Counter-Strike 2. On the first launch, the cursor was still visible while in match even though it shouldn't be and no matter in what direction or how fast I move my mouse, I keep looking down and spinning very fast. I tried relaunching CS2 and that gave me a different outcome: everything was inverted and a bit slow. Moving mouse to the left resulted in the cursor moving right and vice versa, same for up and down. On another relaunch, the cursor was moving correctly, it wasn't inverted, but it was limited to a small space, almost like it is limited to the top left part of the monitor. Downgrading to 3.10.24 through paccache fixed the problem for me.

**Comment by xhenner** [comment]
Same issue on cyberpunk 2077

**Comment by PadTrick** [comment]
same for me.

**Comment by HenrikDK2** [comment]
Same for me, it used to work before last update, now every game doesn't grab the cursor when gamescope is used

**Comment by tobiasjakobi** [comment]
I'm probably seeing the same issue here. I recently updated from v3.14.24 to v3.15.5. Noticed that there are now some games where the mouse input is no longer working. Examples are SOMA and Resident Evil 8.

The mouse still works in the game menu, i.e. when the game shows a mouse cursor which you can move around. But the mouse no longer generates any input when being in-game.

I bisected and this is the bad commit:
https://github.com/ValveSoftware/gamescope/commit/69610ec52429fecbe94c4c042cc42ab43e0491f8

Well, this one is about relative mouse movements, so this bisection result is sensible I guess. I've also double-checked by reverting this specific commit on top of v3.14.28 -- which also works.

I'm using gamescope in conjunction with the DRM backend (greeter starts gamescope and inside a Steam session), so `--force-grab-cursor` does not help me here.

**Comment by tobiasjakobi** [comment]
Please disregard my last comment. I wrote bullshit.

The problems I'm seeing with SOMA are not related to the commit I posted above. SOMA does have mouse issues, but they happen sporadically and right at the start of the game, i.e. where the protagonist is waking up from a dream because of his ringing phone. In this situation the mouse look is restricted, i.e. I can't freely look around but I'm restricted to a small portion of the screen.

This restriction is by game design. The issue that sporadically happens is that the restricted area appears to have zero size. The game does receive input from the mouse, but view is immediately forced back to the original point. It's like a really strong rubberbanding effect going on.

This results in normal mouse movement to appear to have no effect at all. Only if the movement is fast, small movement is seen, but, like I said above, immediately corrected by the game engine.

Hence I'm not sure if we're really seeing a Gamescope bug here. The problem also usually goes away after restarting the game once or twice.

Resident Evil 8 on the other hand is a different thing, and I'm going to open a separate issue for that one.

**Comment by HurricanePootis** [comment]
I am still having this issue with CS2 on Gamescope 3.15.11; my mouse moves inverted and is hitting "bounds" with CS2.

**Comment by iraizo** [comment]
Yeah, can't update gamescope due to this, still stuck on `3.14.24-1`

**Comment by bljordan** [comment]
Still broken on `3.15.5`

**Comment by benbuch** [comment]
This is how it work currently for me.

https://github.com/user-attachments/assets/64e45992-764e-41dd-9338-d17caf4253c6

**Comment by ProjectSynchro** [comment]
What this kind of stinks of is something related to this line in the gamescope log when setting a custom resolution:

`[gamescope] [Info]  edid: Patching res 800x1280 -> 2560x1440` 

Is it possible that this pre-patched screen resolution is making it into the game somehow? I also notice this issue with Halo Infinite.

**Comment by cncz42** [comment]
If you load the steam overlay via
`LD_PRELOAD="" gamescope -f --  env LD_PRELOAD="$HOME/.local/share/Steam/ubuntu12_64/gameoverlayrenderer.so" %command%`
you can fix it by opening and closing it

**Comment by SaphiraKai** [comment]
I'm still having this issue on `3.15.15`

**Comment by PadTrick** [comment]
me too, i'm still using 3.14.22 until this issue is fixed.

**Comment by incognico** [comment]
Same here. I'm on KDE Plasma 6.3 (wayland). Using `--force-grab-cursor ` fixes this for me, tho I get the following console messages in CS2:

![Image](https://github.com/user-attachments/assets/d55d4ec4-e979-4924-9894-ca14a2405f60)

Also I've noticed I can't type an underscore `_`. It always results in `-`, like the shift key is being ignored.

**Comment by PadTrick** [comment]
> Same here. I'm on KDE Plasma 6.3 (wayland). Using `--force-grab-cursor` fixes this for me.
> 
> Also I've noticed I can't type an underscore `_`. It always results in `-`, like the shift key is being ignored.

yes for me also. cant type uppercase at all and for that reason i cant type ?!"§$% etc. very annoying (qwertz) keyboard.
my first thought was it was switched to qwerty, but it isnt.

being not able to type correctly is not that bad as the mouse issue, but still very annoying.
still using the downgraded version until this issues are fixed

**Comment by kisak-valve** [comment]
Hello @incognico, @PadTrick, you're looking for #1740 instead of this issue report.

**Comment by PadTrick** [comment]
question: Why is something hidden as "off topic" if it's an explanation of a new problem that happens if you use the new version with --force-grab-cursor ?

**Comment by biggestsonicfan** [comment]
As this still appears to be an issue, you can build 3.14.24 from source with included dependencies from a specific commit in [OpenSUSE's gamescope git repo](https://src.opensuse.org/pool/gamescope/src/commit/9dae11d8f4343ebd669b2ca12eccd43c561f08efeceae0b3fcf79c5c2271cc8b). `--force-grab-cursor` works but isn't a solution as it causes other issues.

**Comment by GreenSwede** [comment]
This is still an issue with version 3.16.4 when attempting to play Team Fortress 2. Mouse goes upside down and hits the window boundaries when aiming, there are no issues with actually moving the actual mouse cursor in menus. In-game resolution is set to 2560x1440 and it runs in fullscreen.

Steam launch options:
`gamescope  --mangoapp --adaptive-sync -f -W 2560 -H 1440 -r 400 -- gamemoderun %command% -novid -console -nostartupsound +sensitivity 1.6`

**Comment by Vuks69** [comment]
Similar issue in BeamNG - menus work normally, but looking around in first person view on foot has the X- and Y-axis inverted and limited (normally you could just spin and spin in one direction along the X-axis).

Using 3 screens with the game in the middle, if that matters.

**Comment by ManGuyNY** [comment]
> This is still an issue with version 3.16.4 when attempting to play Team Fortress 2. Mouse goes upside down and hits the window boundaries when aiming, there are no issues with actually moving the actual mouse cursor in menus. In-game resolution is set to 2560x1440 and it runs in fullscreen.
> 
> Steam launch options: `gamescope --mangoapp --adaptive-sync -f -W 2560 -H 1440 -r 400 -- gamemoderun %command% -novid -console -nostartupsound +sensitivity 1.6`

Same issue in Half-Life: Opposing Force. Turning "Raw Mouse Input" off fixes the in-game mouse issues, but causes the mouse to be completely invisible in the menus.

**Comment by Krakty** [comment]
I think i hit a related issue running multiple gamescope instances nested on Wayland — cursor position goes stale   when a locked pointer constraint rejects motion under --force-grab-cursor. Submitted a fix in #2100
